### PR TITLE
Moveit py fixes

### PR DESCRIPTION
**Pull request by JafarAbdi**
The main fix here is to be able to change the PlanningScene's current_state. Previously, we would get `RuntimeError: Unable to cast from non-held to held instance (T& to Holder<T>) (compile in debug mode for type information)` because getCurrentState is a reference and pybind can't convert it to a shared_ptr of a RobotState. I made all functions accept a pointer to the class to emulate member functions

The other fix is making current_state an internal_reference.

Other than that, I add some new bindings, updated the stubs, and added `__copy__` & `__deepcopy__` to RobotState and PlanningScene

### CODE REVIEW

File: moveit_py/src/moveit/moveit_core/collision_detection/world.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace moveit_py
+{
+namespace bind_collision_detection
+{
+void init_world(py::module& m)
+{
+  py::module collision_detection = m.def_submodule("collision_detection");
+
+  py::class_<collision_detection::World, collision_detection::WorldPtr>(m, "World").def(py::init<>());
```
Comment (peterdavidfagan) [inline moveit_py/src/moveit/moveit_core/collision_detection/world.cpp:47]: Should we add a TODO to include more methods for interacting with this object from Python? 

My understanding is this is currently introduced in order to support other methods.

Comment (JafarAbdi) [reply moveit_py/src/moveit/moveit_core/collision_detection/world.cpp:47]: Yes, you're right. The only reason why I added it is to be able to construct a planning scene from a robot model.

**Review by peterdavidfagan — APPROVED** [review]
Thanks for these changes and improvements to the Python library @JafarAbdi. I need to finish testing the changes but I have viewed each file and LGTM.

File: moveit_py/src/moveit/moveit_core/planning_scene/planning_scene.cpp [hunk 2]
```diff
@@ -81,6 +81,8 @@ void init_planning_scene(py::module& m)
       		     Representation of the environment as seen by a planning instance. The environment geometry, the robot geometry and state are maintained.
       		     )")
 
+      .def(py::init<const moveit::core::RobotModelConstPtr&, const collision_detection::WorldPtr&>(),
```
[the remark this answers is not in the document]
Comment (JafarAbdi) [reply moveit_py/src/moveit/moveit_core/planning_scene/planning_scene.cpp:84]: Yeah, I think it's a valid use case. I don't want to initialize the whole MoveItPy object to do simple collision checking and forward kinematics stuff

**Pull request merged**